implementations/elixir-jsv/lib/bowtie_jsv.ex

```diff
@@ -0,0 +1,92 @@
+defmodule BowtieJSV do
+  @dialects [
+    "http://json-schema.org/draft-07/schema#",
+    "https://json-schema.org/draft/2020-12/schema"
+  ]
+
+  def main(_argv) do
+    debug("starting")
+    loop_lines(%{})
+    debug("finished")
+  end
+
+  def debug(msg) do
+    IO.puts(:stderr, msg)
+  end
+
+  def debug_value(value, label) do
+    IO.inspect(:stderr, value, label: label)
+  end
+
+  defp loop_lines(state) do
+    debug("loop")
+
+    case IO.read(:stdio, :line) do
+      :eof ->
+        debug("EOF received")
+
+      line ->
+        {:reply, resp, state} = handle_raw_command(line, state)
+        IO.puts(JSON.encode!(resp))
+        loop_lines(state)
+    end
+  end
+
+  defp handle_raw_command(json, state) do
+    with {:ok, cmd} <- JSON.decode(json) do
+      handle_command(cmd, state)
+    end
+  end
+
+  defp handle_command(%{"cmd" => "start", "version" => 1}, state) do
+    start_response = %{
+      version: 1,
+      implementation: %{
+        language: "elixir",
+        name: "jsv",
+        source: "https://github.com/lud/jsv",
+        homepage: "https://github.com/lud/jsv",
+        issues: "https://github.com/lud/jsv/issues",
+        dialects: @dialects
+      }
+    }
+
+    {:reply, start_response, state}
+  end
+
+  defp handle_command(%{"cmd" => "stop"}, _state) do
+    System.halt()
+  end
+
+  defp handle_command(%{"cmd" => "dialect", "dialect" => dialect}, state) do
+    true = dialect in @dialects
```

Comment:
(Again with apologies for not knowing any Elixir), what does this line do?

Is `true` not a keyword in Elixir (i.e. is this assigning a variable called true)?

It's a bug if Bowtie ever sends you a dialect you don't support, so generally there's no need to check this in the harness. What this response is meant to mean is simply "does your implementation support a schema like `{"type": "integer"}` if I tell you externally "that's a draft 7 schema, interpret it as such".

(So I'm suspicious if `true` isn't literally always "return `true` statically", since from below it would seem you do support externally identifying schema dialects).

Reply:
Yeah I should remove this line by now, it is indeed a check to see if the given dialect is in the list. I did this because I do URL normalization, and that trailing `#` in the draft 7 metaschema was causing problems. But it's fixed by now.

(For technical details, `true` is an atom, not a variable indeed, so we are not assigning a value to `true` here (though that sounds like a lot of fun haha), it is meant to crash if the right-hand expression is not equal to `true`.)

Reply:
> For technical details, true is an atom, not a variable indeed, so we are not assigning a value to true here (though that sounds like a lot of fun haha), it is meant to crash if the right-hand expression is not equal to true

Aha! Interesting, I was hoping it was more something like that, interesting.